test/manual/mongodb_oidc.test.ts

```diff
@@ -0,0 +1,450 @@
+            const token = await readFile(`${process.env.OIDC_TOKEN_DIR}/test_user2`, {
+              encoding: 'utf8'
+            });
+            return { accessToken: token };
+          };
+          // - Create a client with a url of the form
+          // ``mongodb://test_user2@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred``
+          // and the OIDC request callback.
+          before(function () {
+            client = new MongoClient(
+              'mongodb://test_user2@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred',
+              {
+                authMechanismProperties: {
+                  REQUEST_TOKEN_CALLBACK: requestCallback
+                }
+              }
+            );
+            collection = client.db('test').collection('test');
+          });
+
+          after(async () => {
+            client?.close();
+          });
+
+          // - Perform a ``find`` operation.
+          // - Clear the cache.
+          it('successfully authenticates', async function () {
+            const doc = await collection.findOne();
+            expect(doc).to.equal(null);
+          });
+        });
+
+        context('when using aws', function () {
+          let client;
+          let collection;
+          const testTokenFile = process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+
+          after(async () => {
+            process.env.AWS_WEB_IDENTITY_TOKEN_FILE = testTokenFile;
+            client?.close();
+          });
+
+          before(async () => {
+            // - Set the ``AWS_WEB_IDENTITY_TOKEN_FILE`` environment variable to the location
+            // of the ``test_user2`` generated token file.
+            process.env.AWS_WEB_IDENTITY_TOKEN_FILE = `${process.env.OIDC_TOKEN_DIR}/test_user2`;
+            // - Create a client with a url of the form
+            // ``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&authMechanismProperties=DEVICE_NAME:aws&directConnection=true&readPreference=secondaryPreferred``.
+            client = new MongoClient(
+              'mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&authMechanismProperties=DEVICE_NAME:aws&directConnection=true&readPreference=secondaryPreferred'
+            );
+            collection = client.db('test').collection('test');
+          });
+
+          // - Perform a ``find`` operation.
+          it('successfully authenticates', async function () {
+            const doc = await collection.findOne();
+            expect(doc).to.equal(null);
+          });
+        });
+      });
+
+      context('when not providing a user', function () {
+        it('fails on option parsing', async function () {
+          expect(async () => {
+            new MongoClient(
+              'mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred'
+            );
+          }).to.throw;
```

Comment:
```suggestion
          }).to.throw();
```

The `throw` property must be called to execute the assertion.

For example, this will pass:

```typescript
it('throws', function() {
	expect(()=> {}).to.throw;
});
```

Reply:
Fixed.